src/Library/demos/Preferences Window/main.blp

```diff
@@ -0,0 +1,160 @@
+using Gtk 4.0;
+using Adw 1;
+
+Adw.PreferencesWindow pref_window {
+  default-width: 800;
+  default-height: 600;
+  title: "My App Preferences";
+
+  Adw.PreferencesPage appearance_page {
+    title:"Appearance";
+    icon-name:"brush-monitor-symbolic";
+
+    Adw.PreferencesGroup{
+      title:"Color Settings";
+      description:"Change the color-scheme of the application.";
+
+
+      Adw.ActionRow{
+        title: "Use Dark Mode";
+        activatable-widget: dm_switch;
+
+        [suffix]
+        Switch dm_switch{
+          halign: center;
+          valign: center;
+        }
+      }
+    }
+
+    Adw.PreferencesGroup {
+      title: "Text Settings";
+      description: "Customize the appearance of text in the application.";
+      //
+
+      Adw.ActionRow {
+        title: "Font Size";
+
+        [suffix]
+        Gtk.SpinButton{
+          halign: center;
+          valign: center;
+          adjustment:
+          Gtk.Adjustment size_adj {
+            lower: 5;
+            upper: 999;
+            step-increment: 1;
+            value: 11;
+          };
+        }
+      }
+
+      Adw.ActionRow {
+        title: "Font Color";
+
+        [suffix]
+        Gtk.ColorDialogButton{
+          halign: center;
+          valign: center;
+          dialog: ColorDialog {};
```

Comment:
is dialog needed? I would expect this to work out of the box

Reply:
![image](https://github.com/sonnyp/Workbench/assets/103920890/307cee41-48f8-4693-9c65-37fd0a32c6c5)
If I don't include a dialog i get this error